src/etc/inc/captiveportal.inc

```diff
@@ -2511,6 +2549,23 @@ function portal_allow($clientip, $clientmac, $username, $password = null, $attri
 					$sessionid, // sessionid
 					time());  // start time
 			}
+			if (xmlrpc_sync_get_details($syncip, $syncport, $syncuser, $syncpass, isset($existing_sessionid))) {
+				// $existing_sessionid prevent carp loop : only forward the connection to the other node if we generated the sessionid by ourselves
+				$rpc_client = new pfsense_xmlrpc_client();
+				$rpc_client->setConnectionData($syncip, $syncport, $syncuser, $syncpass);
+				$rpc_client->set_noticefile("CaptivePortalUserSync");
+				$arguments = array('clientip' => $clientip,
+				'clientmac' => $clientmac,
+				'username' => $username,
+				'password' => $password,
+				'attributes' => $attributes,
+				'allow_time' => $allow_time,
+				'authmethod' => $authmethod,
+				'context' => $context,
+				'sessionid' => $sessionid);
+
+				$rpc_client->xmlrpc_method('captive_portal_sync', array('op' => 'connect_user', 'zone' => $cpzone, 'user' => base64_encode(serialize($arguments))));
```

Comment:
See previous recommendations about making this easier to read/using shorter lines